### PR TITLE
added a 404 page with a new route

### LINKED ISSUE #24: [Feature]: Adding a 404 page
### Is your feature request related to a problem? Please describe.

Adding a custom 404 page can help users navigate to a certain page if they entered into a wrong page. 

### Describe the solution you'd like

I would create a custom animated 404 page that will negate the user frustration and can redirect them back to the homepage 

### Alternatives

_No response_

### Additional Context

Can you assign me this issue under VSOC 24

### Are you interested in contributing?

- [X] I'd like to work on this issue

### Code of Conduct

- [X] I have used the search function to check if an issue already exists
- [X] I agree to follow this project's Code of Conduct

### Which program are you contributing in?

Not applicable

### PR DESCRIPTION
**Pull request by Mehak-Mattoo**
### Description

I have added a custom 404 route and a page

### Fixes

issue closes #24 

### Type of PR

 Feature enhancement

### Summary

 I have made a additional route in App.jsx which will catch all mistyped links and bring the user to a 404 page, from where they can make their way back to homepage

### Additional Notes

Could you increase the issue difficulty to medium as it was not a minor task.

### Checklist

- [* ] I have read and followed the [Pull Requests and Issues guidelines](https://github.com/digitomize/digitomize/blob/main/PR_GUIDELINES.md#pull-requests-and-issues).
- [ *] The code has been properly linted and formatted using `npm run lint:fix` and `npm run format:fix`.
- [ *I have tested the changes thoroughly before submitting this pull request.
- [* ] I have provided relevant issue numbers, snapshots, and videos after making the changes.
- [* ] I have not borrowed code without disclosing it, if applicable.
- [* ] This pull request is not a Work In Progress (WIP), and only completed and tested changes are included.
- [ *] I have tested these changes locally.
- [* ] My code follows the project's style guidelines.
- [* ] I have updated the documentation accordingly.
- [* ] This PR has a corresponding issue in the issue tracker.

## Summary by CodeRabbit

- **New Features**
  - Introduced a custom 404 Not Found page with an image, title, description, and a link to navigate back to the home page.
  
- **Style**
  - Added responsive styles for the 404 Not Found page to ensure proper layout and design across different screen sizes.

### CODE REVIEW

File: client/src/components/ui/NotFound/NotFound.css
Comment (Latish705) [inline client/src/components/ui/NotFound/NotFound.css:1]: we are using tailwind in this project so use tailwind for css

**Review by Latish705 — CHANGES_REQUESTED** [review]
@Mehak-Mattoo done excellent work just need to write tailwind css and provide screenshot of page or video of implementation

File: client/src/components/ui/NotFound/NotFound.css
Comment (Mehak-Mattoo) [reply client/src/components/ui/NotFound/NotFound.css:1]: Sorry, I have used tailwind now and removed the css file.

**Review by Latish705 — APPROVED** [review]
@Mehak-Mattoo show screen shot

**Comment by Mehak-Mattoo** [comment]
I had put  it my PR but again here is it

**Pull request merged**